c_src/membrane_ffmpeg_swscale_plugin/pixel_format_converter/converter.c

```diff
@@ -0,0 +1,122 @@
+    goto end;
+  }
+
+  state->width = width;
+  state->height = height;
+  state->srcFormat = input_fmt;
+  state->dstFormat = output_fmt;
+
+  // Allocate memory for both the input and outputframes
+  int src_image_size =
+      av_image_alloc(state->src_data, state->src_linesize, state->width,
+                     state->height, state->srcFormat, 1);
+  state->dst_image_size =
+      av_image_alloc(state->dst_data, state->dst_linesize, state->width,
+                     state->height, state->dstFormat, 1);
+
+  if (src_image_size < 0 || state->dst_image_size < 0) {
+    sws_freeContext(state->sws_context);
+    res = create_result_error(env, "memory_allocation_failed");
+    goto end;
+  }
+
+  res = create_result_ok(env, state);
+end:
+  unifex_release_state(env, state);
+  return res;
+}
+
+UNIFEX_TERM process(UnifexEnv *env, State *state, UnifexPayload *payload) {
+  UNIFEX_TERM ret;
+
+  // copy input to av_image
+  av_image_fill_arrays(state->src_data, state->src_linesize, payload->data,
+                       state->srcFormat, state->width, state->height, 1);
+
+  // perform the conversion
+  if (sws_scale(state->sws_context, (const uint8_t *const *)state->src_data,
+                (const int *)state->src_linesize, 0, state->height,
+                state->dst_data, state->dst_linesize) < 0) {
+    ret = process_result_error(env, "scaling_failed");
+    goto end;
+  }
+
+  UnifexPayload output_payload;
+  unifex_payload_alloc(env, UNIFEX_PAYLOAD_BINARY, state->dst_image_size,
+                       &output_payload);
+  // copy output from av_frame to buffer
+  if (av_image_copy_to_buffer(output_payload.data, state->dst_image_size,
+                              (const uint8_t *const *)state->dst_data,
+                              (const int *)state->dst_linesize,
+                              state->dstFormat, state->width, state->height,
+                              1) < 0) {
+    ret = process_result_error(env, "copy_to_buffer_failed");
+  } else {
+    ret = process_result_ok(env, &output_payload);
+  }
+  unifex_payload_release(&output_payload);
+end:
+  return ret;
+}
+
+void handle_destroy_state(UnifexEnv *env, State *state) {
+  UNIFEX_UNUSED(env);
+  if (state->sws_context) {
+    sws_freeContext(state->sws_context);
+  }
+}
+
+enum AVPixelFormat string_to_AVPixelFormat(char *format) {
+  if (strcmp(format, "I420") == 0) {
+    return AV_PIX_FMT_YUV420P;
+  } else if (strcmp(format, "I422") == 0) {
+    return AV_PIX_FMT_YUV422P;
+  } else if (strcmp(format, "I444") == 0) {
+    return AV_PIX_FMT_YUV444P;
+  } else if (strcmp(format, "RGB") == 0) {
+    return AV_PIX_FMT_RGB24;
+  } else if (strcmp(format, "RGBA") == 0) {
+    return AV_PIX_FMT_RGBA;
+  } else if (strcmp(format, "NV12") == 0) {
+    return AV_PIX_FMT_NV12;
+  } else if (strcmp(format, "NV21") == 0) {
+    return AV_PIX_FMT_NV21;
+  } else if (strcmp(format, "YV12") == 0) {
+    return AV_PIX_FMT_YUVA420P;
```

Comment:
Yeah, good point. Quick google search indicates that to process it with FFmpeg you need to indicate `yuv420p` and shuffle the planes. This might complicate things, but I think more research is needed

Reply:
As discussed, I just removed support for this pixel format